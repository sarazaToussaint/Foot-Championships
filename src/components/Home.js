import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { BiRightArrowCircle } from 'react-icons/bi';
import { Link } from 'react-router-dom';
import { fetchLeagues } from './redux/league';
import Leagues from './Leagues';
import classes from './Home.module.css';

const Home = () => {
  const leagues = useSelector((state) => state.leagues);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchLeagues());
  }, []);

  return (
    <>
      <div className={classes.mainContainer}>
        <p className={classes.total}>
          Total Leagues
          &nbsp;
          Available:
          {' '}
          {leagues.length}
        </p>
        <div className="lowerContainer">
          <Link to="/Leagues">
            <BiRightArrowCircle
              className={classes.mainIcon}
              style={{
                width: '25px',
                height: '25px',
                color: '#fff',
                border: 'none',
                zIndex: '4',
                opacity: '1',
              }}
            />
          </Link>
        </div>
      </div>
      <Leagues />
    </>
  );
};

export default Home;
