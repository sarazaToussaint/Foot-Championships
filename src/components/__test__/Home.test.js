import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configStore';
import Home from '../Home';

describe('Home test', () => {
  it('renders leagues', () => {
    const app = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
