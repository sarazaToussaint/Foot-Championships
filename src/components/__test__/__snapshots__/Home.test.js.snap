// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test renders leagues 1`] = `
Array [
  <div
    className="mainContainer"
  >
    <p
      className="total"
    >
      Total Leagues Â  Available:
       
      0
    </p>
    <div
      className="lowerContainer"
    >
      <a
        href="/Leagues"
        onClick={[Function]}
      >
        <svg
          className="mainIcon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "border": "none",
              "color": "#fff",
              "height": "25px",
              "opacity": "1",
              "width": "25px",
              "zIndex": "4",
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
          />
          <path
            d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
          />
        </svg>
      </a>
    </div>
  </div>,
  <div
    className="grid-container"
  >
    <div>
      <h2>
        WELCOME TO TOP 20 FIFA LEAGUES
      </h2>
    </div>
    <div
      className="leagues"
    />
  </div>,
]
`;
